mood entry UI improvements (#96)Feature/frontend mood entry UI 96

Nobie12 opened the pull request:
### Summary
This PR implements frontend improvements for the mood journaling feature as described in Issue #96.

### Changes
1. **Entry Card Display**
   - Show title if present
   - Show note snippet if title is missing
   - Truncate long text and ensure responsive layout
   - Handle empty edge cases gracefully

2. **Mood Screen Layout**
   - Display 4 moods per row and 2 rows per screen
   - Add scroll arrows for navigating moods
   - Implement smooth scroll logic
   - Tested across screen sizes for responsiveness

3. **Add Title to Journal Entries**
   - Added title input field in NewEntryForm modal
   - Send title along with entry payload (guest + logged-in)
   - Display saved title in EntryCard
   - Confirmed edit/update flow still works as expected

### Notes
- Branch: `feature/frontend-mood-entry-ui-96`
- Linked Issue: #96
- Thoroughly tested


closes ( #96  )

Nobie120 committed:
feat(entry-card): update EntryCard to show title, snippet fallback, truncation, and edge case handling (#96)

Nobie120 committed:
feat(mood-screen): display moods in grid with scroll arrows and smooth scroll logic (#96)

Nobie120 committed:
feat(entries): add title field to journal entries and display in EntryCard (#96)

Nobie120 committed:
chore: cleanup debug logs and unused code (#96)

STARR44 reviewed (commented):
![Code](https://github.com/user-attachments/assets/d91f8da8-2666-4c3a-8df4-2e56ce584a84)


Hey Ken! Good work here 🙌

I've attached the particular issue in NewEntryForm.jsx so you can see the full context.

In the file, around line 72, inside the setTimeout for shouldShowKindness():

If shouldShowKindness() is false and shouldShowLoginPrompt() is also false, navigate("/entries") never runs — users may think submission failed.

Also, entries can currently be submitted with just a title. Could we revert to:

return alert("Please pick a mood or write something!");


so that only a note or mood is required?

Nobie120 committed:
Fix entry form validation to require a note or mood (title alone no longer allowed)
 (#96)

Nobie120 committed:
feat: lock mood selection and improve journal flow

- Added click lock to MoodSelector (prevents multiple selections)
- Handled guest vs logged-in moods with fallback
- Added error and empty state messages with CSS
- Restored scroll lock wrapper in MoodJournalPage

Nobie120 committed:
feat: lock mood selection and improve journal flow

- Added click lock to MoodSelector (prevents multiple selections)
- Handled guest vs logged-in moods with fallback
- Added error and empty state messages with CSS
- Restored scroll lock wrapper in MoodJournalPage

Nobie120 committed:
Previously, moods API included a hardcoded fallback list.
This fallback has been removed so authenticated users always rely on backend data.
The full static moods list is now stored in guestMoods for guest users only.

Nobie120 committed:
refinement of the mood screen

Nobie120 committed:
removed the fallback from moods api (#96)

Nobie12 committed:
Merge branch 'dev' into feature/frontend-mood-entry-ui-96

STARR44 reviewed (commented):
Good improvements
